src/sidebar/services/test/frame-sync-test.js

```diff
@@ -111,8 +113,11 @@ describe('FrameSyncService', () => {
         },
 
         connectFrame(frame) {
+          const otherFrames = this.getState().frames.filter(
+            f => f.id !== frame.id
+          );
```

Comment:
This makes `connectFrame` do an upsert of the passed frame based on its ID, matching the real `connectFrame` action.